src/os/bluestore/StupidAllocator.cc

```diff
@@ -198,6 +198,40 @@ int StupidAllocator::allocate(
   return 0;
 }
 
+int StupidAllocator::alloc_extents(
+  uint64_t want_size, uint64_t alloc_unit, uint64_t max_alloc_size,
+  int64_t hint, std::vector<AllocExtent> *extents, int *count)
+{
+  uint64_t allocated_size = 0;
+  uint64_t offset = 0;
+  uint32_t length = 0;
+  int res = 0;
+
+  if (max_alloc_size == 0) {
+    max_alloc_size = want_size;
+  }
+
+  ExtentList block_list = ExtentList(extents, 1, max_alloc_size);
+
+  while (allocated_size < want_size) {
+    res = allocate(MIN(max_alloc_size, (want_size - allocated_size)),
+       alloc_unit, hint, &offset, &length);
+    if (res != 0) {
+      /*
+       * Allocation failed.
+       */
+      assert(0);
```

Comment:
You do a release and no assert for the similar case in BitMapAllocator. Suggest to have uniform behavior..